graph/random_walker.py

```diff
@@ -0,0 +1,38 @@
+from functools import cache
+from typing import Dict
+import networkx as nx
+import numpy as np
+from random import randrange, random
+from graph.network_edge_list import NetworkEdgeList
+
+
+class RandomWalker:
+    """
+    Future potential optimization: load all edges into a single array as described in Section 3.3 of
+    Eksombatchai et al. (2018) and use offsets to randomly access. Current method is very slow and
+    dynamically allocates neighbor dicts (bad bad bad)
+    """
+    network_edge_list: NetworkEdgeList
+
+    def __init__(self, network_edge_list):
+        self.network_edge_list = network_edge_list
```

Comment:
Will this actually result in a network_edge_list being constructed, or is this set to the class? this may be my lack of Python knowledge

Reply:
The `network_edge_list` object is passed into the constructor, meaning that some other code constructed it and is now passing the reference to the `RandomWalker` so that it can use it.